packages/lexical/src/LexicalEvents.ts

```diff
@@ -609,7 +611,16 @@ function onBeforeInput(event: InputEvent, editor: LexicalEditor): void {
             isSelectionAnchorSameAsFocus &&
             !hasSelectedAllTextInNode &&
             selectedNodeCanInsertTextAfter;
-          if (!shouldLetBrowserHandleDelete) {
+          // Check if selection is collapsed and if the previous node is a decorator node
+          // If so, the browser will not be able to handle the deletion
+          let browserAbleToDelete = true;
+          if (selection.isCollapsed()) {
+            const prevNode = $getAdjacentNode(selection.anchor, true);
+            if ($isDecoratorNode(prevNode)) {
+              browserAbleToDelete = false;
+            }
+          }
+          if (!(shouldLetBrowserHandleDelete && browserAbleToDelete)) {
```

Comment:
It would make more sense to declare `shouldLetBrowserHandleDelete` with `let` so you can mutate it instead of having two variables here, or define a function that implements the logic that you could include when initially defining `shouldLetBrowserHandleDelete`.

e.g. 

```suggestion
          // Check if selection is collapsed and if the previous node is a decorator node
          // If so, the browser will not be able to handle the deletion
          if (shouldLetBrowserHandleDelete && selection.isCollapsed()) {
            shouldLetBrowserHandleDelete = !$isDecoratorNode($getAdjacentNode(selection.anchor, true));
          }
          if (!shouldLetBrowserHandleDelete) {
```